lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -79,26 +80,50 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
 
     TargetSnapshot target;
     try {
-      target = targetSelector.select();
+      target = getTarget();
     } catch (Exception e) {
       throw new PauserException("Failed to find the target pods to pause.", e);
     }
 
-    RequestCoordinator coordinator = getRequestCoordinator(target);
+    RequestCoordinator coordinator;
+    try {
+      coordinator = getRequestCoordinator(target);
+    } catch (Exception e) {
+      throw new PauserException("Failed to initialize the coordinator.", e);
+    }
 
-    coordinator.pause(true, maxPauseWaitTime);
+    Instant startTime;
+    Instant endTime;
+    try {
+      coordinator.pause(true, maxPauseWaitTime);
 
-    Instant startTime = Instant.now();
+      startTime = Instant.now();
 
-    Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
+      Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
 
-    Instant endTime = Instant.now();
+      endTime = Instant.now();
 
-    unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+      unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+
+    } catch (Exception e) {
+      try {
+        unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+      } catch (PauserException ex) {
+        throw new PauserException("unpauseWithRetry() method failed twice.", e);
+      } catch (Exception ex) {
+        throw new PauserException(
+            "unpauseWithRetry() method failed twice due to unexpected exception.", e);
+      }
+      throw new PauserException(
+          "The pause operation failed for some reason. However, the unpause operation succeeded"
+              + " afterward. Currently, the scalar products are running with the unpause status."
+              + " You should retry the pause operation to ensure proper backup.",
+          e);
+    }
```

Comment:
This is the main update of this PR.

Previously, there was no exception handling. So, if some error occurred, this library didn't try to run `unpause()`. This might cause the Scalar products to keep the pause state. In other words, it might cause like system down.

To address such issue, I added the exception handling here. In this implementation, if some errors occur, this library tries to run the `unpause()` operation.